src/vm/runners/builtin_runner/mod.rs

```diff
@@ -289,7 +289,10 @@ impl BuiltinRunner {
             })
             .collect::<Vec<_>>();
 
-        let n = div_floor(offsets.len(), cells_per_instance as usize);
+        let n = match offsets.iter().max() {
+            Some(x) => div_floor(*x, cells_per_instance as usize) + 1,
+            None => 0,
+        };
```

Comment:
A `map_or` could shorten this match to `offsets.iter().max().map_or(0, |v| div_floor(*x, cells_per_instance as usize) + 1)`